packages/s3-store/index.ts

```diff
@@ -235,31 +238,36 @@ export class S3Store extends DataStore {
       .on('chunkFinished', ({path, size: partSize}) => {
         pendingChunkFilepath = null
 
+        const partNumber = currentPartNumber++
+        const chunkNumber = currentChunkNumber++
+
+        offset += partSize
+
+        const isFirstChunk = chunkNumber === 0
```

Comment:
If you increment `currentChunkNumber` before this check, won't it always be `> 0`?